src/components/Table/TableHead/TableHead.jsx

```diff
@@ -147,9 +150,43 @@ const TableHead = ({
   closeMenuText,
   lightweight,
   i18n,
+  hasResize,
 }) => {
   const filterBarActive = activeBar === 'filter';
-
+  const utilsVar ={
+    startX: 0,
+    pressed: false,
+    index : 0
+  }
+  const [state, setState] = useState({TableHeader:[]});
+  const elementsRef = useRef(ordering.map(() => createRef()));
+  useEffect(() => {
+    const nextWidth = elementsRef.current.map(
+      ref => ref.current.getBoundingClientRect().width
+    );
+    setState(nextWidth);
+  }, []);
+  
+  const onMouseMovecallback = (e) => {
+    if (utilsVar.pressed) {
+      setState(psd => ({ ...psd, 
+        [utilsVar.index]: state[utilsVar.index] + (e.x - utilsVar.startX),
+        [utilsVar.index + 1 ]: state[utilsVar.index + 1] - (e.x - utilsVar.startX),
+      }));
+    }
+  };
+  const onMouseUpcallback = () => {
+    if (utilsVar.pressed) {
+      utilsVar.pressed = false;
+    }
+  }
+  const onMouseDownCallback = (e, index) =>{
+    utilsVar.pressed = true;            
+    utilsVar.startX = e.clientX;  
+    utilsVar.index = index;  
+    document.addEventListener('mousemove', onMouseMovecallback, true);
```

Comment:
Can we not use these react [mouse events](https://reactjs.org/docs/events.html#mouse-events) instead of adding listeners to the document. If we must put them here we should also be removing these event handlers when the component un-mounts.